File: ios-app/NailsDashIOS/Sources/Features/Appointments/AppointmentModels.swift
import Foundation

struct AppointmentCreateRequest: Encodable {
    let store_id: Int
    let service_id: Int
    let technician_id: Int?
    let appointment_date: String
    let appointment_time: String
    let notes: String?
}

struct AppointmentGroupGuestCreateRequest: Encodable {
    let service_id: Int
    let technician_id: Int?
    let notes: String?
    let guest_name: String?
    let guest_phone: String?
}

struct AppointmentGroupCreateRequest: Encodable {
    let store_id: Int
    let appointment_date: String
    let appointment_time: String
    let host_service_id: Int
    let host_technician_id: Int?
    let host_notes: String?
    let guests: [AppointmentGroupGuestCreateRequest]
}

struct AppointmentGroupResponseDTO: Decodable {
    let group_id: Int
    let group_code: String?
    let host_appointment: AppointmentDTO
    let guest_appointments: [AppointmentDTO]
}

struct AppointmentCancelRequest: Encodable {
    let cancel_reason: String?
}

struct AppointmentRescheduleRequest: Encodable {
    let new_date: String
    let new_time: String
}

struct AppointmentDTO: Decodable, Identifiable {
    let id: Int
    let order_number: String?
    let store_id: Int
    let service_id: Int
    let technician_id: Int?
    let appointment_date: String
    let appointment_time: String
    let status: String
    let notes: String?
    let store_name: String?
    let store_address: String?
    let service_name: String?
    let service_price: Double?
    let service_duration: Int?
    let technician_name: String?
    let created_at: String?
    let cancel_reason: String?
}

struct TechnicianDTO: Decodable, Identifiable {
    let id: Int
    let store_id: Int
    let name: String
    let is_active: Int
    let avatar_url: String?
}

struct ServiceDTO: Decodable, Identifiable {
    let id: Int
    let store_id: Int
    let name: String
    let price: Double
    let duration_minutes: Int
    let is_active: Int
}

struct TechnicianAvailableSlotDTO: Decodable, Hashable {
    let start_time: String
    let end_time: String
    let duration_minutes: Int
}

File: android-app/app/src/main/java/com/nailsdash/android/data/model/AppointmentModels.kt
package com.nailsdash.android.data.model

data class AppointmentCreateRequest(
    val store_id: Int,
    val service_id: Int,
    val technician_id: Int? = null,
    val appointment_date: String,
    val appointment_time: String,
    val notes: String? = null,
)

data class AppointmentGroupGuestCreateRequest(
    val service_id: Int,
    val technician_id: Int? = null,
    val notes: String? = null,
    val guest_name: String? = null,
    val guest_phone: String? = null,
)

data class AppointmentGroupCreateRequest(
    val store_id: Int,
    val appointment_date: String,
    val appointment_time: String,
    val host_service_id: Int,
    val host_technician_id: Int? = null,
    val host_notes: String? = null,
    val guests: List<AppointmentGroupGuestCreateRequest> = emptyList(),
)

data class AppointmentGroupResponse(
    val group_id: Int,
    val group_code: String? = null,
    val host_appointment: Appointment,
    val guest_appointments: List<Appointment>,
)

data class AppointmentCancelRequest(
    val cancel_reason: String? = null,
)

data class AppointmentRescheduleRequest(
    val new_date: String,
    val new_time: String,
)

data class Appointment(
    val id: Int,
    val order_number: String? = null,
    val store_id: Int,
    val service_id: Int,
    val technician_id: Int? = null,
    val appointment_date: String,
    val appointment_time: String,
    val status: String,
    val order_amount: Double? = null,
    val notes: String? = null,
    val store_name: String? = null,
    val store_address: String? = null,
    val service_name: String? = null,
    val service_price: Double? = null,
    val service_duration: Int? = null,
    val technician_name: String? = null,
    val review_id: Int? = null,
    val completed_at: String? = null,
    val created_at: String? = null,
    val cancel_reason: String? = null,
)

data class Technician(
    val id: Int,
    val store_id: Int,
    val name: String,
    val is_active: Int,
    val avatar_url: String? = null,
)

data class ServiceItem(
    val id: Int,
    val store_id: Int,
    val name: String,
    val price: Double,
    val duration_minutes: Int,
    val is_active: Int,
)

data class TechnicianAvailableSlot(
    val start_time: String,
    val end_time: String,
    val duration_minutes: Int,
)
